File: app/src/main/java/com/katdmy/android/unocount/PlayerAdapter.java
package com.katdmy.android.unocount;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.Switch;
import android.widget.TextView;

import java.util.List;

public class PlayerAdapter extends BaseAdapter {

    private final LayoutInflater mInflater;
    private List<Player> mPlayers;
    private OnItemClickListener mClickListener;
    private OnCheckedChangeListener mCheckedListener;

    public interface OnItemClickListener {
        void onDeleteClick(Player player);
    }

    void setOnItemClickListener(OnItemClickListener clickListener) {
        mClickListener = clickListener;
    }

    public interface OnCheckedChangeListener {
        void onCheckedChange(String playerName, boolean isChecked);
    }

    void setOnCheckedChangeListener(OnCheckedChangeListener checkedListener) {
        mCheckedListener = checkedListener;
    }

    PlayerAdapter(Context context) {
        mInflater = LayoutInflater.from(context);
    }

    void setPlayers(List<Player> players) {
        mPlayers = players;
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        if (mPlayers != null)
            return mPlayers.size();
        else return 0;
    }

    @Override
    public Player getItem(int position) {
        return mPlayers.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View itemView = convertView;
        if (itemView == null) {
            itemView = mInflater.inflate(R.layout.player_item, parent, false);
        }

        Player player = getItem(position);

        Switch activeSwitch = itemView.findViewById(R.id.activeSwitch);
        TextView nameTextView = itemView.findViewById(R.id.nameTextView);
        ImageView deleteImage = itemView.findViewById(R.id.deleteImage);

        activeSwitch.setChecked(player.getActive());
        nameTextView.setText(player.getName());

        activeSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (mCheckedListener != null)
                mCheckedListener.onCheckedChange(player.getName(), isChecked);
        });

        deleteImage.setOnClickListener(v -> {
            if (mClickListener != null)
                mClickListener.onDeleteClick(mPlayers.get(position));
        });

        return itemView;
    }
}

File: app/src/main/java/com/katdmy/android/unocount/ui/player/PlayerAdapter.kt
package com.katdmy.android.unocount.ui.player

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.switchmaterial.SwitchMaterial
import com.katdmy.android.unocount.R
import com.katdmy.android.unocount.domain.player.model.Player

class PlayerAdapter(
        private val deleteClickListener: (playerName: String) -> Unit,
        private val checkedChangeListener: (playerName: String, isChecked: Boolean) -> Unit
) : RecyclerView.Adapter<PlayerAdapter.PlayerViewHolder>() {
    private var players: List<Player> = emptyList()

    inner class PlayerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val activeSwitch: SwitchMaterial = itemView.findViewById(R.id.activeSwitch)
        private val nameTextView: TextView = itemView.findViewById(R.id.nameTextView)
        private val deleteImage: ImageView = itemView.findViewById(R.id.deleteImage)

        fun onBind(player: Player) {
            activeSwitch.isChecked = player.active
            nameTextView.text = player.name
            activeSwitch.setOnCheckedChangeListener { _: CompoundButton?, isChecked: Boolean -> checkedChangeListener(player.name, isChecked) }
            deleteImage.setOnClickListener { deleteClickListener(player.name) }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlayerViewHolder {
        val view: View = LayoutInflater.from(parent.context)
                .inflate(R.layout.player_item, parent, false)
        return PlayerViewHolder(view)
    }

    override fun onBindViewHolder(holder: PlayerViewHolder, position: Int) {
        holder.onBind(players[position])
    }

    override fun getItemCount(): Int {
        return players.size
    }

    fun setPlayers(newPlayers: List<Player>) {
        players = newPlayers
        notifyDataSetChanged()
    }
}